special-case/src/main/java/com/iluwatar/specialcase/InvalidUser.java

```diff
@@ -42,6 +42,8 @@ public InvalidUser(String userName) {
 
   @Override
   public void show() {
-    LOGGER.info("Invalid user: " + userName);
+    if (LOGGER.isInfoEnabled()) {
+      LOGGER.info(String.format("Invalid user: %s", userName));
+    }
```

Comment:
I would leave out the call `LOGGER.isInfoEnabled()`. The logger object already calls it internally and I think there is no reason to use it if we are not doing something very expensive inside the if-condition. Check this elsewhere within the changes as well.

Reply:
Fixed.